core/src/main/java/com/oath/oak/InternalOakMap.java

```diff
@@ -1161,22 +1213,30 @@ public T next() {
 
         public T next() {
 
-            Map.Entry<ByteBuffer, Handle> pair = advance();
-            Handle handle = pair.getValue();
+            Map.Entry<ByteBuffer, Slice> pair = advance(true, true);
             ByteBuffer serializedKey = pair.getKey();
-            if (handle == null) {
+            Slice valueSlice = pair.getValue();
+            if (valueSlice == null) {
                 return null;
             }
-            handle.readLock();
-            if (handle.isDeleted()) {
-                handle.readUnLock();
+            ValueUtils.ValueResult res = ValueUtils.lockRead(valueSlice);
```

Comment:
I assume all that is going to change after merge with Eran... Eran introduced all those transformations going inside the handle, and you might have something parallel that can be used here...